libs/langchain/langchain/retrievers/self_query/supabase.py

```diff
@@ -0,0 +1,96 @@
+from typing import Any, Dict, Tuple
+
+from langchain.chains.query_constructor.ir import (
+    Comparator,
+    Comparison,
+    Operation,
+    Operator,
+    StructuredQuery,
+    Visitor,
+)
+
+
+class SupabaseVectorTranslator(Visitor):
+    """Translate Langchain filters to Supabase PostgREST filters."""
+
+    allowed_operators = [Operator.AND, Operator.OR]
+    """Subset of allowed logical operators."""
+    allowed_comparators = [
+        Comparator.EQ,
+        Comparator.NE,
+        Comparator.GT,
+        Comparator.GTE,
+        Comparator.LT,
+        Comparator.LTE,
+        Comparator.CONTAIN,
+        Comparator.LIKE,
+    ]
+    """Subset of allowed logical comparators."""
+
+    metadata_column = "metadata"
+
+    def _format_value(self, value: Any, comparator: Comparator) -> str:
+        if comparator == Comparator.CONTAIN:
+            return f"%{value}%"
+
+        return value
+
+    def _map_comparator(self, comparator: Comparator) -> str:
+        """
+        Maps Langchain comparator to PostgREST comparator:
+
+        https://postgrest.org/en/stable/references/api/tables_views.html#operators
+        """
+        return {
+            Comparator.EQ: "eq",
+            Comparator.NE: "neq",
+            Comparator.GT: "gt",
+            Comparator.GTE: "gte",
+            Comparator.LT: "lt",
+            Comparator.LTE: "lte",
+            Comparator.CONTAIN: "like",
+            Comparator.LIKE: "like",
+        }.get(comparator, comparator)
```

Comment:
should we let it error if comparators not present?

Reply:
Good call, updated 👍